test/common/formatter/substitution_formatter_test.cc

```diff
@@ -2814,6 +2822,7 @@ TEST(SubstitutionFormatterTest, DynamicMetadataFieldExtractor) {
     std::string val = formatter.format(stream_info).value();
     EXPECT_TRUE(val.find("\"test_key\":\"test_value\"") != std::string::npos);
     EXPECT_TRUE(val.find("\"test_obj\":{\"inner_key\":\"inner_value\"}") != std::string::npos);
+    EXPECT_TRUE(val.find("\"test_lst\":[\"item0\",4.2]") != std::string::npos);
```

Comment:
Frustrating because you're copying the structure of nearby existing code but the nearby existing code is yuck: could you instead do
```
EXPECT_THAT(val, HasSubstr(R"("test_lst":["item0",4.2]")"));
```
(this is more readable and, in the case that the test fails, would show the string and expectation, vs. the existing code on test failure would just say something useless along the lines of "expected true but was false")

Fixing the adjacent ones in the same way optional.